### PR TITLE
MM-23725 add team restore

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary

This PR adds the `team restore` command to mmctl which restores previously archived teams.

Note, this PR is based off https://github.com/mattermost/mmctl/pull/164 which adds `team modify`.  Once that PR is merged, the diff for this one will get much smaller.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-23725

### CODE REVIEW

File: commands/team_test.go [hunk 1]
```diff
@@ -688,3 +688,219 @@ func (s *MmctlUnitTestSuite) TestSearchTeamCmd() {
...
+		s.client.
+			EXPECT().
+			RestoreTeam(teamID).
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
+		s.client.
+			EXPECT().
+			GetTeam(teamName, "").
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
```
Comment (mgdelacroix) [inline commands/team_test.go:870]: nit: I think mocks are easier to understand if they appear in the same order they are called by the corresponding code

File: commands/team_test.go [hunk 2]
```diff
@@ -688,3 +688,219 @@ func (s *MmctlUnitTestSuite) TestSearchTeamCmd() {
...
+			EXPECT().
+			RestoreTeam(teamID).
+			Return(nil, &model.Response{Error: mockError}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetTeam(teamName, "").
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
```
Comment (mgdelacroix) [inline commands/team_test.go:899]: Same nit here

File: commands/team_test.go (hunk 1 above)
Comment (wiggin77) [reply commands/team_test.go]: Fixed here, spot below, and several other places that were pre-existing.

File: commands/team_test.go [hunk 3]
```diff
@@ -825,3 +830,84 @@ func (s *MmctlUnitTestSuite) TestModifyTeamsCmd() {
...
+			EXPECT().
+			RestoreTeam(teamID).
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetTeam(teamName, "").
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
```
Comment (mgdelacroix) [inline commands/team_test.go:876]: Same mock order comment here

File: commands/team_test.go [hunk 4]
```diff
@@ -825,3 +830,84 @@ func (s *MmctlUnitTestSuite) TestModifyTeamsCmd() {
...
+			EXPECT().
+			GetTeamByName(teamName, "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetTeam(teamName, "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
```
Comment (mgdelacroix) [inline commands/team_test.go:852]: Same mock order comment here

File: commands/team_test.go (hunk 1 above)
Comment (mgdelacroix) [reply commands/team_test.go]: Saw that you fixed it retroactively for the other PR as well, thanks a lot!! :smile:

File: commands/team_test.go (hunk 3 above)
Comment (wiggin77) [reply commands/team_test.go]: Right, I did the fix before merging from master.  Thanks.

**Pull request merged**